sdk/core/azure-core/src/test/java/com/azure/core/http/rest/PagedIterableTest.java

```diff
@@ -277,6 +278,44 @@ public void streamFindFirstOnlyRetrievesOnePage() throws InterruptedException {
         assertEquals(1, pageRetriever.getGetCount() - DEFAULT_PAGE_COUNT);
     }
 
+    private <S> Function<PagedResponse<S>, PagedResponse<S>> mapPagedResponse(Function<S, S> mapper) {
+        return pagedResponse -> new PagedResponseBase<HttpRequest, S>(pagedResponse.getRequest(),
+            pagedResponse.getStatusCode(),
+            pagedResponse.getHeaders(),
+            pagedResponse.getValue().stream().map(mapper).collect(Collectors.toList()),
+            pagedResponse.getContinuationToken(),
+            null);
+    }
+
+    @Test
+    public void streamFindFirstWithMapPage() throws InterruptedException {
+        int numberOfPages = 5;
+        createPagedResponse(numberOfPages);
+
+        AtomicInteger pageGetCount = new AtomicInteger(0);
+        PagedFlux<Integer> pagedFlux = new PagedFlux<>(() -> {
+            pageGetCount.getAndIncrement();
+            return pagedResponses.isEmpty() ? Mono.empty() : Mono.just(pagedResponses.get(0));
+        },
+            continuationToken -> {
+            pageGetCount.getAndIncrement();
+            return getNextPage(continuationToken, pagedResponses);
+        });
+
+        PagedIterable<Integer> pagedIterable = new PagedIterable<>(pagedFlux);
+
+        pagedIterable = pagedIterable.mapPage(Function.identity());
```

Comment:
If the change in `PagedFlux` is not applied, this line will cause the case retrieving all paged (5), instead of only first page (1).